subsys/bluetooth/mesh/large_comp_data_srv.c

```diff
@@ -59,24 +59,54 @@ static int handle_large_comp_data_get(struct bt_mesh_model *model, struct bt_mes
 	LOG_DBG("page %u offset %u", page, offset);
 
 	bt_mesh_model_msg_init(&rsp, OP_LARGE_COMP_DATA_STATUS);
-
-	if (page != 0U) {
+	if (page >= 130U && IS_ENABLED(CONFIG_BT_MESH_COMP_PAGE_2) &&
```

Comment:
Can all this code be moved into `access.c` and reused in `cfg_srv.c` with `offset` == 0?

Reply:
I will investigate the possibillity. I think I will push all other resolved comments first, then push a fix for this later so that we can keep the momentum up on the PR.

Reply:
by later you mean in a separate PR?

Reply:
No, I've added a suggestion commit now

Reply:
No, sorry, I don't like the new function. It adds a lot of models specific dependencies to the access layer. What I was worried about is the bunch of if-else cases for page selection and then reading the page depending on the page number and dirty flag. But this change now create the opposite situation. Lets drop the last commit and do the refactoring some time later.

Reply:
I changed it to only include the page parsing. At least a minor improvment IMO.